src/utils/transformer-utils.ts

```diff
@@ -1,188 +1,320 @@
+};
+
+export const createComponentBlock = (
+  guid: string,
+  reference: Reference
+): PortableTextComponent => {
+  return {
+    _type: "component",
+    _key: guid,
+    component: reference,
+  };
+};
+
+export const compose = <T>(
+  firstFunction: (argument: T) => T,
+  ...functions: Array<(argument: T) => T>
+) =>
+  functions.reduce(
+    (previousFunction, nextFunction) => (value) =>
+      previousFunction(nextFunction(value)),
+    firstFunction
+  );
+
+/**
+ * Renders a portable text table to HTML or a Vue virtual DOM node.
+ * 
+ * @param {PortableTextTable} table - The portable text table object to render.
+ * @param {(value: PortableTextBlock[]) => string} resolver - A function that resolves 
+ *        the content of each cell in the table.
+ * @param {Function} [vueRenderFunction] - An optional Vue render function for generating 
+ *        virtual DOM nodes. If provided, the table is rendered as Vue nodes, otherwise 
+ *        as an HTML string.
+ * @returns {string | VueNode} The rendered table as an HTML string or Vue virtual DOM node.
+ */
+export const resolveTable = (
+  table: PortableTextTable,
+  resolver: (value: PortableTextBlock[]) => string,
+  vueRenderFunction?: Function
+) => {
+  const renderCell = (cell: PortableTextTableCell) => {
+    const cellContent = resolver(cell.content);
+    return vueRenderFunction
+      ? vueRenderFunction("td", {}, cellContent)
+      : `<td>${cellContent}</td>`;
+  };
+
+  const renderRow = (row: PortableTextTableRow) => {
+    const cells = row.cells.map(renderCell);
+    return vueRenderFunction
+      ? vueRenderFunction("tr", {}, cells)
+      : `<tr>${cells.join("")}</tr>`;
+  };
+
+  const renderRows = () =>
+    vueRenderFunction
+      ? table.rows.map(renderRow)
+      : table.rows.map(renderRow).join("");
+
+  return vueRenderFunction
+    ? vueRenderFunction("table", {}, renderRows())
+    : `<table><tbody>${renderRows()}</tbody></table>`;
+};
+  
+/**
+ * Resolves an image object to HTML or a Vue virtual DOM node.
+ * 
+ * @param {PortableTextImage} image - The portable text image object to be rendered.
+ * @param {(image: PortableTextImage) => string} [resolver] - An optional resolver function 
+ *        that returns the image as an HTML string. If not provided, a default HTML 
+ *        representation is used.
+ * @param {Function} [vueRenderFunction] - An optional Vue render function for generating 
+ *        virtual DOM nodes. If provided, the image is rendered as a Vue node, otherwise 
+ *        as an HTML string.
+ * @returns {string | VueNode} The resolved image as an HTML string or Vue virtual DOM node.
+ */
+export const resolveImage = (
+  image: PortableTextImage,
+  resolver?: (image: PortableTextImage) => string,
+  vueRenderFunction?: Function
+) => {
+  if (vueRenderFunction)
+    return vueRenderFunction("img", {
+      src: image.asset.url,
+      alt: image.asset.alt || "",
+    });
+
+  return resolver
+    ? resolver(image)
```

Comment:
It seems that in most of the resolvers you either call the provided resolver or use a custom implementation. I think it would be better to have the default implementation separated into separate resolvers and simply always call some resolver function. Either one that the user provided or provide the default resolver.

Reply:
split and resolved in https://github.com/kontent-ai/rich-text-resolver-js/pull/39/commits/5bcc38a09f412a4928b1ecc3f5e351b122d9b0ec